Content.Server/Zombies/ZombieSystem.cs

```diff
@@ -156,24 +159,102 @@ private void DoGroan(EntityUid uid, ActiveZombieComponent component)
             component.LastDamageGroanCooldown = component.GroanCooldown;
         }
 
+        private void OnZombieInit(EntityUid ent, ZombieComponent component, ComponentStartup _)
+        {
+            // Brute
+            component.HealingDamageSpecifier.DamageDict["Brute"] = -10;
+            component.HealingDamageSpecifier.DamageDict["Pierce"] = -10;
+            component.HealingDamageSpecifier.DamageDict["Blunt"] = -10;
+
+            // Burn
+            component.HealingDamageSpecifier.DamageDict["Heat"] = -10;
+            component.HealingDamageSpecifier.DamageDict["Shock"] = -10;
+            component.HealingDamageSpecifier.DamageDict["Cold"] = -10;
+
+            // Zombies do not receive any Toxic or Airloss damage
+        }
+
+        /// <summary>
+        /// Heals zombie if it is in bad state and not dead
+        /// </summary>
+        private void TryHealZombie(EntityUid ent, ZombieComponent component)
+        {
+            if (!TryComp<MobStateComponent>(ent, out var state) || !TryComp<DamageableComponent>(ent, out var damage))
+            {
+                return;
+            }
+
+
+            if (state.CurrentState == DamageState.Critical)
+            {
+                _damageable.TryChangeDamage(ent, component.HealingDamageSpecifier, true);
+                return;
+            }
+
+            if (state.CurrentState != DamageState.Alive)
+                return;
+            var dmg = damage.TotalDamage;
+
+            // Always zero, I guess?
+            var aliveThreshold = 0f;
+            var critStateData = _state.GetEarliestCriticalState(state, dmg);
+            var nextThreshold = FixedPoint2.Zero;
+            if (critStateData != null)
+            {
+                nextThreshold = critStateData.Value.threshold;
+            }
+            else
+            {
+                var deadStateData = _state.GetEarliestDeadState(state, dmg);
+                if (deadStateData == null)
+                {
+                    return;
+                }
+
+                nextThreshold = deadStateData.Value.threshold;
+            }
+
+
+            // Healing to 80% of HP. Not too big and not too small.
```

Comment:
Personally healing to near full seems too much for existing. 

They should heal enough to get out of crit and perhaps be encouraged to eat the deceased to get above 50%. Although iirc all zombie kills are conversions which is another issue.

Reply:
My bad commenting. They actually heal to have 80% of damage required to drop critical.

Reply:
My bad commenting. They actually heal to 80% of damage required to drop critical.